samples/secure-fraud-detection/src/main/java/io/quarkiverse/langchain4j/sample/FraudDetectionContentRetriever.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkiverse.langchain4j.sample;
+
+import java.util.List;
+
+import org.eclipse.microprofile.jwt.JsonWebToken;
+import org.jboss.logging.Logger;
+
+import dev.langchain4j.rag.content.Content;
+import dev.langchain4j.rag.content.retriever.ContentRetriever;
+import dev.langchain4j.rag.query.Query;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+@ApplicationScoped
+public class FraudDetectionContentRetriever implements ContentRetriever {
+	private static final Logger log = Logger.getLogger(FraudDetectionContentRetriever.class);
+
+	@Inject
+	TransactionRepository transactionRepository;
+
+	@Inject
+	CustomerRepository customerRepository;
+
+    @Override
+    public List<Content> retrieve(Query query) {
+        JsonWebToken idToken = (JsonWebToken)query.metadata().chatMemoryId();
```

Comment:
I'm puzzled as to what the content retriever should actually do. .With this, I just get `java.lang.ClassCastException: class java.lang.String cannot be cast to class org.eclipse.microprofile.jwt.JsonWebToken (java.lang.String is in module java.base of loader 'bootstrap'; org.eclipse.microprofile.jwt.JsonWebToken is in unnamed module of loader io.quarkus.bootstrap.classloading.QuarkusClassLoader @78b1cc93)` on this line

Reply:
It is converted to String somewhere in transit, looking into it

Reply:
Hi @jmartisk @geoand, OK, I'm just doing a simple  User name/email conversion from String to the `UserNameAndEmail` bean and it is good for now. Working on Readme next and it should be ready for review

Reply:
@sberyozkin I am still getting the same error and the method, even if the error were fixed, would return an empty list anyway. Do I need to change something to get it working?